components/sync_manager/src/syncmanager.udl

```diff
@@ -48,12 +48,15 @@ interface SyncEngineSelection {
     Some(sequence<string> engines);
 };
 
-enum SyncReason {
+enum RustSyncReason {
     "Scheduled",
     "User",
     "PreSleep",
     "Startup",
     "EnabledChange",
+    "Backgrounded",
+    "DidLogin",
```

Comment:
Android treats `DidLogin` and `EnabledChange` as the same - https://github.com/mozilla-mobile/firefox-android/blob/5a961111addca08dcaa402d2a79b05920c04e731/android-components/components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/sync/Types.kt#L42. I wonder if we should just do the same here? Otherwise, we should probably get a PR/issue up for Android to use this new one (and given the above link, that seems like it would be a trivial change for them)

Reply:
Also `DidLogin` is also mainly used by the old sync code so I think I'm going to remove it to and map `SyncReason.DidLogin` to `RustSyncReason.EnabledChange`.